<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1217</width>
    <height>829</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QHBoxLayout" name="horizontalLayout_3">
    <item>
     <layout class="QVBoxLayout" name="verticalLayout_2">
      <item>
       <layout class="QVBoxLayout" name="verticalLayout">
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_2">
          <item>
           <widget class="QLabel" name="label_2">
            <property name="minimumSize">
             <size>
              <width>100</width>
              <height>0</height>
             </size>
            </property>
            <property name="text">
             <string>Number of iterations</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QSpinBox" name="spinBox_2">
            <property name="maximum">
             <number>15000</number>
            </property>
            <property name="value">
             <number>500</number>
            </property>
           </widget>
          </item>
         </layout>
        </item>
       </layout>
      </item>
      <item>
       <layout class="QHBoxLayout" name="CPUSerialGroup">
        <item>
         <widget class="QPushButton" name="CPUserial">
          <property name="text">
           <string>CPU serial</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="CPUserialLabel">
          <property name="text">
           <string/>
          </property>
          <property name="alignment">
           <set>Qt::AlignCenter</set>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <layout class="QHBoxLayout" name="CPUParallelGroup">
        <item>
         <widget class="QPushButton" name="CPUparallel">
          <property name="text">
           <string>CPU parallel</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="CPUparallelLabel">
          <property name="text">
           <string/>
          </property>
          <property name="alignment">
           <set>Qt::AlignCenter</set>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <layout class="QHBoxLayout" name="GPUGroup">
        <item>
         <widget class="QPushButton" name="GPU">
          <property name="text">
           <string>GPU</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="GPULabel">
          <property name="text">
           <string/>
          </property>
          <property name="alignment">
           <set>Qt::AlignCenter</set>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout">
        <item>
         <widget class="QPushButton" name="AVXserial">
          <property name="text">
           <string>AVX serial</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="AVXserialLabel">
          <property name="text">
           <string/>
          </property>
          <property name="alignment">
           <set>Qt::AlignCenter</set>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_4">
        <item>
         <widget class="QPushButton" name="AVXparallel">
          <property name="text">
           <string>AVX parallel</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="AVXparallelLabel">
          <property name="text">
           <string/>
          </property>
          <property name="alignment">
           <set>Qt::AlignCenter</set>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <widget class="QPushButton" name="pushButton">
        <property name="maximumSize">
         <size>
          <width>100</width>
          <height>16777215</height>
         </size>
        </property>
        <property name="text">
         <string>Reset</string>
        </property>
       </widget>
      </item>
      <item>
       <spacer name="verticalSpacer">
        <property name="orientation">
         <enum>Qt::Vertical</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>20</width>
          <height>40</height>
         </size>
        </property>
       </spacer>
      </item>
     </layout>
    </item>
    <item>
     <widget class="Canvas" name="widget" native="true">
      <property name="minimumSize">
       <size>
        <width>1024</width>
        <height>768</height>
       </size>
      </property>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1217</width>
     <height>20</height>
    </rect>
   </property>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <customwidgets>
  <customwidget>
   <class>Canvas</class>
   <extends>QWidget</extends>
   <header>canvas.h</header>
   <container>1</container>
   <slots>
    <signal>changeIterationSpinBox(int)</signal>
    <signal>changeWorkersSpinbox(int)</signal>
    <signal>resetLabels()</signal>
    <signal>setCPUSerialLabel(qint64)</signal>
    <signal>setCPUParallelLabel(qint64)</signal>
    <signal>setGPULabel(qint64)</signal>
    <signal>setAVXSerialLabel(qint64)</signal>
    <signal>setAVXParallelLabel(qint64)</signal>
    <slot>changeNumOfWorkers(int)</slot>
    <slot>changeNumOfIterations(int)</slot>
    <slot>reset()</slot>
    <slot>calculateCPUSerial()</slot>
    <slot>calculateCPUParallel()</slot>
    <slot>calculateGPU()</slot>
    <slot>calculateAVXSerial()</slot>
    <slot>calculateAVXParallel()</slot>
   </slots>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections>
  <connection>
   <sender>spinBox_2</sender>
   <signal>valueChanged(int)</signal>
   <receiver>widget</receiver>
   <slot>changeNumOfIterations(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>174</x>
     <y>54</y>
    </hint>
    <hint type="destinationlabel">
     <x>193</x>
     <y>147</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>widget</sender>
   <signal>changeIterationSpinBox(int)</signal>
   <receiver>spinBox_2</receiver>
   <slot>setValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>270</x>
     <y>121</y>
    </hint>
    <hint type="destinationlabel">
     <x>174</x>
     <y>54</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>widget</sender>
   <signal>resetLabels()</signal>
   <receiver>MainWindow</receiver>
   <slot>resetTimeLabels()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>220</x>
     <y>285</y>
    </hint>
    <hint type="destinationlabel">
     <x>110</x>
     <y>318</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>widget</sender>
   <signal>setCPUSerialLabel(qint64)</signal>
   <receiver>MainWindow</receiver>
   <slot>setCPUSerialLabel(qint64)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>304</x>
     <y>528</y>
    </hint>
    <hint type="destinationlabel">
     <x>108</x>
     <y>483</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>widget</sender>
   <signal>setCPUParallelLabel(qint64)</signal>
   <receiver>MainWindow</receiver>
   <slot>setCPUParallelLabel(qint64)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>251</x>
     <y>371</y>
    </hint>
    <hint type="destinationlabel">
     <x>111</x>
     <y>403</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>widget</sender>
   <signal>setGPULabel(qint64)</signal>
   <receiver>MainWindow</receiver>
   <slot>setGPULabel(qint64)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>226</x>
     <y>256</y>
    </hint>
    <hint type="destinationlabel">
     <x>109</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>MainWindow</sender>
   <signal>calculateCPUSerial()</signal>
   <receiver>widget</receiver>
   <slot>calculateCPUSerial()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>109</x>
     <y>220</y>
    </hint>
    <hint type="destinationlabel">
     <x>234</x>
     <y>212</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>MainWindow</sender>
   <signal>calculateCPUParallel()</signal>
   <receiver>widget</receiver>
   <slot>calculateCPUParallel()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>88</x>
     <y>180</y>
    </hint>
    <hint type="destinationlabel">
     <x>196</x>
     <y>182</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>MainWindow</sender>
   <signal>calculateGPU()</signal>
   <receiver>widget</receiver>
   <slot>calculateGPU()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>57</x>
     <y>245</y>
    </hint>
    <hint type="destinationlabel">
     <x>205</x>
     <y>241</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>MainWindow</sender>
   <signal>calculateAVXSerial()</signal>
   <receiver>widget</receiver>
   <slot>calculateAVXSerial()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>51</x>
     <y>356</y>
    </hint>
    <hint type="destinationlabel">
     <x>211</x>
     <y>368</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>MainWindow</sender>
   <signal>calculateAVXParallel()</signal>
   <receiver>widget</receiver>
   <slot>calculateAVXParallel()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>53</x>
     <y>549</y>
    </hint>
    <hint type="destinationlabel">
     <x>197</x>
     <y>542</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>widget</sender>
   <signal>setAVXSerialLabel(qint64)</signal>
   <receiver>MainWindow</receiver>
   <slot>setAVXSerialLabel(qint64)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>232</x>
     <y>585</y>
    </hint>
    <hint type="destinationlabel">
     <x>120</x>
     <y>587</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>widget</sender>
   <signal>setAVXParallelLabel(qint64)</signal>
   <receiver>MainWindow</receiver>
   <slot>setAVXParallelLabel(qint64)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>221</x>
     <y>613</y>
    </hint>
    <hint type="destinationlabel">
     <x>147</x>
     <y>628</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton</sender>
   <signal>clicked()</signal>
   <receiver>widget</receiver>
   <slot>reset()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>104</x>
     <y>223</y>
    </hint>
    <hint type="destinationlabel">
     <x>284</x>
     <y>229</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <signal>calculateCPUSerial()</signal>
  <signal>calculateCPUParallel()</signal>
  <signal>calculateGPU()</signal>
  <signal>calculateAVXSerial()</signal>
  <signal>calculateAVXParallel()</signal>
  <slot>reset()</slot>
  <slot>resetTimeLabels()</slot>
  <slot>setCPUSerialLabel(qint64)</slot>
  <slot>setCPUParallelLabel(qint64)</slot>
  <slot>setGPULabel(qint64)</slot>
  <slot>setAVXSerialLabel(qint64)</slot>
  <slot>setAVXParallelLabel(qint64)</slot>
 </slots>
</ui>
